flow/project.py

```diff
@@ -895,25 +946,31 @@ def _generate_id(self, job, operation_name=None, index=0):
         if max_len < len(job_op_id):
             raise ValueError("Value for MAX_LEN_ID is too small ({}).".format(self.MAX_LEN_ID))
 
+        if len(jobs) > 1:
+            aggregate_id = str(jobs[0])[:8]+'-'+str(jobs[-1])[:8]
+        else:
+            aggregate_id = str(jobs[0])[:8]
+
         separator = getattr(project._environment, 'JOB_ID_SEPARATOR', '/')
-        readable_name = '{project}{sep}{job}{sep}{op_string}{sep}{index:04d}{sep}'.format(
-                    sep=separator,
-                    project=str(project)[:12],
-                    job=str(job)[:8],
-                    op_string=op_string[:12],
-                    index=index)[:max_len]
+        readable_name = '{project}{sep}{jobs}{sep}{op_string}' \
+                        '{sep}{index:04d}{sep}'.format(
```

Comment:
Is there a reason this is split into two lines now?

Reply:
I changed the way we printed `id` but then realized that it may not be something we should change in this PR. Forgot to change it back to the original state, will do it now.
Thank you for pointing out